index.js

```diff
@@ -76,42 +91,61 @@ BemConfig.prototype.get = function() {
 };
 
 /**
- * Resolves config for given level
- * @param {String} levelPath - level path
- * @returns {Promise}
+ * Returns merged config synchronously
+ * @returns {Object}
  */
-BemConfig.prototype.level = function(levelPath) {
-    const absLevelPath = path.resolve(this._options.projectRoot, levelPath);
+BemConfig.prototype.getSync = function() {
+    return Object.assign.apply(Object, [{}].concat(this.configs(true)));
+};
+
+function getLevelByConfigs(levelPath, options, allConfigs) {
+    const absLevelPath = path.resolve(options.projectRoot, levelPath);
     let levelOpts = { __source: absLevelPath };
 
-    return this.configs().then(all => {
-        for (let i = 0; i < all.length; i++) {
-            let conf = all[i];
-            let levels = conf.levels || {};
+    for (let i = 0; i < allConfigs.length; i++) {
+        let conf = allConfigs[i];
+        let levels = conf.levels || {};
 
-            Object.assign(levelOpts, conf);
+        Object.assign(levelOpts, conf);
 
-            for (let level in levels) {
-                if (level === absLevelPath) {
+        for (let level in levels) {
+            if (level === absLevelPath) {
 
-                    // works like deep extend but overrides arrays
-                    levelOpts = _.mergeWith({}, levels[level], levelOpts, (objValue, srcValue) => {
-                        if (Array.isArray(objValue)) {
-                            return srcValue;
-                        }
-                    });
-                }
+                // works like deep extend but overrides arrays
+                levelOpts = _.mergeWith({}, levels[level], levelOpts, (objValue, srcValue) => {
+                    if (Array.isArray(objValue)) {
+                        return srcValue;
```

Comment:
`(objValue, srcValue) => Array.isArray(objValue) && srcValue || undefined`?
isn't it ugly?